src/Xamarin.Android.Build.Tasks/Utilities/MarshalMethodsClassifier.cs

```diff
@@ -0,0 +1,388 @@
+using System;
+using System.Collections.Generic;
+
+using Java.Interop.Tools.Cecil;
+using Java.Interop.Tools.JavaCallableWrappers;
+using Java.Interop.Tools.TypeNameMappings;
+using Microsoft.Android.Build.Tasks;
+using Microsoft.Build.Utilities;
+using Mono.Cecil;
+
+namespace Xamarin.Android.Tasks
+{
+#if ENABLE_MARSHAL_METHODS
+	public sealed class MarshalMethodEntry
+	{
+		public TypeDefinition TopType             { get; }
```

Comment:
`TopType` may be better named `DeclaringType`?  Not sure what "Top" is supposed to mean.